src/components/FieldOptionEditor.tsx

```diff
@@ -0,0 +1,43 @@
+import * as React from 'react';
+import * as data from '../data';
+
+interface IProps {
+    field: data.IField;
+    registry: data.FieldRegistry;
+    onChange: (field: data.IField) => void;
```

Comment:
I think we should expose the complete form definition when a field changes so the consumer doesn't need to manually update the fields array.